### DISCUSSION TITLE
How do I turn off errors array in production mode?

### DISCUSSION THREAD

**Discussion by pquerner** [typestack/class-validator#2495] [Questions & Answers]
In production mode I'd like to give out no details about what went wrong with the validation.
How can I turn it off?

Validation should still take place, but I don't wish to give out any details about why it failed.

Thanks

**Comment by sunnypatell** [comment]
@pquerner in NestJS, use the `disableErrorMessages` option on `ValidationPipe`:

```ts
app.useGlobalPipes(
  new ValidationPipe({
    disableErrorMessages: process.env.NODE_ENV === 'production',
  })
);
```

this returns a 400 Bad Request with no constraint details in the response body when enabled. validation still runs and rejects invalid input, it just strips the error messages from the response.

if you're not using NestJS, catch the validation errors and return a generic message:

```ts
const errors = await validate(dto);
if (errors.length > 0) {
  throw new HttpException('Validation failed', 400);
}
```

ref: [NestJS ValidationPipe docs](https://docs.nestjs.com/techniques/validation#disable-detailed-errors)

**Comment by usualdesigner** [comment]
class-validator itself doesn't have a 'production mode' switch for this - handle it at the layer that catches the ValidationError array. Run validate() as normal so validation still happens, but in your error handler, if NODE_ENV is production, map the errors to a generic message instead of serializing the full objects back to the client. Keep logging the detailed errors server-side, just don't ship them in the response.
